src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] list;
+    private int size;
+    private int currentCapacity;
```

Comment:
better avoid using extra variable by using `list.length`. creating redundant variables can lead to bugs when once we will forget to update it
```suggestion
```